src/Compilers/Core/Portable/SourceGeneration/GeneratorDriverOptions.cs

```diff
@@ -29,5 +32,24 @@ public GeneratorDriverOptions(IncrementalGeneratorOutputKind disabledOutputs, bo
             DisabledOutputs = disabledOutputs;
             TrackIncrementalGeneratorSteps = trackIncrementalGeneratorSteps;
         }
+
+        /// <summary>
+        /// Creates <see cref="GeneratorDriverOptions"/>.
+        /// </summary>
+        /// <param name="disabledOutputs"></param>
+        /// <param name="trackIncrementalGeneratorSteps"></param>
+        /// <param name="baseDirectory">Absolute path to the base directory used for file paths of generated files.</param>
+        /// <exception cref="ArgumentException"><paramref name="baseDirectory"/> is not an absolute path.</exception>
+        public GeneratorDriverOptions(IncrementalGeneratorOutputKind disabledOutputs = IncrementalGeneratorOutputKind.None, bool trackIncrementalGeneratorSteps = false, string? baseDirectory = null)
+        {
+            if (baseDirectory != null && !PathUtilities.IsAbsolute(baseDirectory))
+            {
+                throw new ArgumentException(nameof(baseDirectory), CodeAnalysisResources.AbsolutePathExpected);
```

Comment:
Looks like message and paramName are swapped in the ArgumentException constructor. Consider using named arguments.